### ISSUE TITLE
Equation compiler needs qualified constructor

### ISSUE THREAD

**Issue by andreasabel** [leanprover/lean3#1709]
Lean version 3.2.0, commit 9a386ae188e8, Release.

This works:
```lean
inductive Ty
| base
| arr (a b : Ty)
open Ty

def cxt := list Ty

inductive Var : Π (g : cxt) (a : Ty), Type
| vz {g a}   : Var (a :: g) a
| vs {g a b} : Var g a → Var (b :: g) a
open Var

inductive Tm : cxt → Ty → Type
| var {g a}   : Var g a → Tm g a
| app {g a b} : Tm g (arr a b) → Tm g a → Tm g b
| abs {g a b} : Tm (a :: g) b → Tm g (arr a b)
open Tm

def Ren (g d : cxt) := Π a, Var d a → Var g a

def liftr {g d a} (r : Ren g d) : Ren (a :: g) (a :: d)
| ._  vz    := vz
| _  (vs x) := vs (r _ x)

def rename : Π {g d} (r : Ren g d) {a}, Tm d a → Tm g a
| g d r _  (var x)            := var (r _ x)
| g d r _  (@app ._ _ ._ t u) := app (rename r t) (rename r u)
| g d r ._ (@Tm.abs ._ _ _ t) := abs (rename (liftr r) t)
```
It fails if the qualification `Tm.` is removed in the last line.
```lean
| g d r ._ (@abs ._ _ _ t)    := abs (rename (liftr r) t)
```
Error is: (35 is last line)
```
    35   9 error           invalid use of inaccessible term, it is not completely fixed by other arguments
   arr ?m_1 ?m_2 (lean-checker)
    35  14 error           invalid pattern, 'abs' is overloaded, and some interpretations may occur in patterns and others not (solution: use fully qualified names) (lean-checker)
    35  14 error           function expected at
   abs (lean-checker)
```

**Comment by Kha** [comment]
This was fixed in 3300eafd391351038cb95ecf5a791fee5c51d736

**Comment by andreasabel** [comment]
Thanks, I'll wait for the release.
